src/main/java/org/alliancegenome/curation_api/dao/base/BaseSQLDAO.java

```diff
@@ -147,18 +163,41 @@ public E merge(E entity) {
 	public E remove(String id) {
 		log.debug("SqlDAO: remove: " + id);
 		E entity = find(id);
-		entityManager.remove(entity);
+
+		try {
+			entityManager.remove(entity);
+			entityManager.flush();
+		} catch (PersistenceException e) {
+			handlePersistenceException(entity, e);
+		}
 		return entity;
 	}
 
 	@Transactional
 	public E remove(Long id) {
 		log.debug("SqlDAO: remove: " + id);
 		E entity = find(id);
-		entityManager.remove(entity);
+
+		try {
+			entityManager.remove(entity);
+			entityManager.flush();
+		} catch (PersistenceException e) {
+			handlePersistenceException(entity, e);
+		}
 		return entity;
 	}
 	
+	private void handlePersistenceException(E entity, Exception e) {
+		ObjectResponse<E> response = new ObjectResponse<E>(entity);
+		if (e.getCause() instanceof ConstraintViolationException) {
+			ConstraintViolationException cve = (ConstraintViolationException) e.getCause();
```

Comment:
Might want to just double check that you are getting the "rootCause" vs any cause it would then list the table and constraint that is violated.

Reply:
I also agree with your slack message this would be better handled in the controller where it could craft a better error message. At this point in the code there is no indication of who the caller is or what happened.